domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt

```diff
@@ -24,11 +30,15 @@ class ApplicationLifecycleObserver @Inject constructor(
   private val learnerAnalyticsLogger: LearnerAnalyticsLogger,
   private val profileManagementController: ProfileManagementController,
   private val oppiaLogger: OppiaLogger,
+  private val performanceMetricsLogger: PerformanceMetricsLogger,
   private val performanceMetricsController: PerformanceMetricsController,
   @LearnerAnalyticsInactivityLimitMillis private val inactivityLimitMillis: Long,
   @BackgroundDispatcher private val backgroundDispatcher: CoroutineDispatcher
 ) : ApplicationStartupListener, LifecycleObserver {
 
+  private var alreadyRunningInForeground = false
+  private var alreadyRunningInBackground = false
```

Comment:
This implies four states, but in reality it should only ever be two (foreground & background), right? Could we consolidate them into a single variable?

Reply:
Removed these states entirely since they weren't needed anymore.